js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            method = 'spotPublicGetTickerBookTicker';
+            //
+            //     [
+            //       {
+            //         "symbol": "AEUSDT",
+            //         "bidPrice": "0.11001",
+            //         "bidQty": "115.59",
+            //         "askPrice": "0.11127",
+            //         "askQty": "215.48"
+            //       },
+            //     ]
+            //
+        } else if (marketType === 'swap') {
+            throw new BadRequest (this.id + ' fetchBidsAsks() is not available for ' + marketType + ' markets');
+        }
+        const response = await this[method] (query);
+        return this.parseTickers (response, symbols);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['spot']) {
+            return await this.createSpotOrder (market, type, side, amount, price, params);
+        } else if (market['swap']) {
+            return await this.createSwapOrder (market, type, side, amount, price, params);
+        }
+    }
+
+    async createSpotOrder (market, type, side, amount, price = undefined, params = {}) {
+        const symbol = market['symbol'];
+        const orderSide = (side === 'buy') ? 'BUY' : 'SELL';
+        // TODO: needs postOnly handing here, possible with LIMIT_MAKER
+        const request = {
+            'symbol': market['id'],
+            'quantity': this.amountToPrecision (symbol, amount),
+            'side': orderSide,
+            'type': type.toUpperCase (),
+        };
+        if (price !== undefined) {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        const clientOrderId = this.safeString (params, 'clientOrderId');
+        if (clientOrderId !== undefined) {
+            request['newClientOrderId'] = clientOrderId;
+            params = this.omit (params, [ 'type', 'clientOrderId' ]);
+        }
+        const response = await this.spotPrivatePostOrder (this.extend (request, params));
+        //
+        // spot
+        //
+        //     {
+        //         "symbol": "BTCUSDT",
+        //         "orderId": "123738410679123456",
+        //         "orderListId": -1
+        //     }
+        //
+        return this.extend (this.parseOrder (response, market), {
+            'side': side,
+            'type': type,
+            'price': price, // TODO: should we do this?
+            'amount': amount,
+        });
+    }
+
+    async createSwapOrder (market, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const symbol = market['symbol'];
+        const openType = this.safeInteger (params, 'openType');
+        if (openType === undefined) {
+            throw new ArgumentsRequired (this.id + ' createSwapOrder() requires an integer openType parameter, 1 for isolated margin, 2 for cross margin');
+        }
```

Comment:
For some reason I could not figure out yet I can't also create swap Orders

Output:

```
mexc3.createOrder (LTC/USDT:USDT, limit, 1, 0.5, 50, [object Object])
fetch Request:
 mexc3 POST https://contract.mexc.com/api/v1/private/order/submit 
RequestHeaders:
RequestBody:
 {"symbol":"LTC_USDT","vol":0.5,"side":1,"type":1,"openType":2,"price":50} 

handleRestResponse:
 mexc3 POST https://contract.mexc.com/api/v1/private/order/submit 200 OK 
ResponseHeaders:
ResponseBody:
 {"success":false,"code":1002,"message":"Contract not allow place order!"} 

InvalidOrder mexc3 {"success":false,"code":1002,"message":"Contract not allow place order!"}
---------------------------------------------------
[InvalidOrder] mexc3 {"success":false,"code":1002,"message":"Contract not allow place order!"}

    at throwBroadlyMatchedException  ../../js/base/Exchange.js:647         throw new broad[broadKey] (message)                                             
    at handleErrors                  ../../js/mexc3.js:3659                this.throwBroadlyMatchedException (this.exceptions['broad'], body, feedback);   
    at                               ../../js/base/Exchange.js:717         const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                  
    at async timeout                 ../../js/base/functions/time.js:203   return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at createSwapOrder               ../../js/mexc3.js:1605                const response = await this[method] (this.extend (request, params));            
    at createOrder                   ../../js/mexc3.js:1492                return await this.createSwapOrder (market, type, side, amount, price, params);  
    at async main                    cli.js:269                            const result = await exchange[methodName] (... args)                            

mexc3 {"success":false,"code":1002,"message":"Contract not allow place order!"}
```

Reply:
According to the info from the exchange techs:

> BTC_USDT ETH_USDT LTC_USDT has not been opened yet

Try any other symbol.

Reply:
We could probably use BadSymbol in this case, cause the order is valid, but the symbol is not tradeable.
I would add more comments here and a map of "not opened" contracts in the options, for a more meaningful exception message and to avoid further confusion among the devs and users.

Reply:
@kroitor should I wait , will you be pushing changes for this in the PR?

Reply:
@ttodua It's ok, i'll get to it later.

Reply:
I haven't been able to find any swap symbols that don't return this error, I think that we should disable `createOrder` for swap, and also remove `options["unavailableContracts"]` for the purpose of avoiding errors/confusion when/if they ever do fix their api and allow the creation of swap orders